meta-resin-common/recipes-containers/docker-disk/files/entry.sh

```diff
@@ -8,9 +8,10 @@ DOCKER_TIMEOUT=20 # Wait 20 seconds for docker to start
 # Default values
 PARTITION_SIZE=${PARTITION_SIZE:=1024}
 
-# Create a BTRFS disk image
-dd if=/dev/zero of=/export/data_disk.img bs=1M count=$PARTITION_SIZE
-mkfs.btrfs --mixed --metadata=single /export/data_disk.img
+# Create sparse file to hold ext4 resin-data partition
+dd if=/dev/zero of=/export/data_disk.img bs=1M count=0 seek=$PARTITION_SIZE
```

Comment:
All `mkfs.*` programs support it one way or another. This is how yocto is doing it as well[1]:

```
IMAGE_CMD_jffs2 = "mkfs.jffs2 --root=${IMAGE_ROOTFS} [...]
IMAGE_CMD_cramfs = "mkfs.cramfs ${IMAGE_ROOTFS}[...]

# fstype is one of [ ext2, ext3, ext4 ]
mkfs.$fstype -F $extra_imagecmd [...] -d ${IMAGE_ROOTFS}
mkfs.btrfs ${EXTRA_IMAGECMD} -r ${IMAGE_ROOTFS}[...]
IMAGE_CMD_squashfs = "mksquashfs ${IMAGE_ROOTFS} [...]
IMAGE_CMD_ubifs = "mkfs.ubifs -r ${IMAGE_ROOTFS} [...]
```

And all these commands are run in a fakeroot environment and so they produce a filesystem image with the correct files and permissions

[1] https://github.com/MentorEmbedded/poky/blob/master/meta/classes/image_types.bbclass